### PR TITLE
[release-0.15] Update references to release v0.15.0

### PR DESCRIPTION
**Pull request by marquiz**
Generated with:

    hack/prepare-release.sh -g 1.21.5 v0.15.0

### CODE REVIEW

**Comment by marquiz** [comment]
Let's wait for #1523 
/hold

File: deployment/base/gc/gc.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ spec:
       serviceAccount: nfd-gc
       containers:
         - name: nfd-gc
-          image: gcr.io/k8s-staging-nfd/node-feature-discovery:master
-          imagePullPolicy: Always
+          image: registry.k8s.io/nfd/node-feature-discovery:v0.15.0
+          imagePullPolicy: IfNotPresent
```
Comment (fmuyassarov) [inline deployment/base/gc/gc.yaml:21]: @marquiz Just curios why is the policy being changed too?

**Review by fmuyassarov — COMMENTED** [review]
/lgtm

**Pull request merged**